typing/ctype.ml

```diff
@@ -2092,7 +2092,7 @@ let expand_trace env trace =
 let deep_occur t0 ty =
   let rec occur_rec ty =
     let ty = repr ty in
-    if ty.level >= lowest_level then begin
+    if ty.level >= t0.level then begin
```

Comment:
Just to check: this is an unrelated optimisation right?

Reply:
Indeed. Since `deep_occur` looks deeply, it seems better to stop as early as we can.